nexus-repository-r/src/test/java/org/sonatype/nexus/repository/r/internal/upload/RUploadHandlerTest.java

```diff
@@ -142,23 +144,41 @@ public void testHandle_unauthorized() throws IOException {
     when(contentPermissionChecker.isPermitted(eq(REPO_NAME), eq(RFormat.NAME), eq(BreadActions.EDIT), any()))
         .thenReturn(false);
 
-    ComponentUpload component = createComponentUpload(PACKAGE_PATH);
+    ComponentUpload component = createComponentUpload(PACKAGE_PATH, PACKAGE_NAME);
+    checkValidationFailed(component, String.format("Not authorized for requested path '%s'", PACKAGE_PATH_FULL));
+  }
+
+  @Test
+  public void testHandleValidationExceptionWrongPath() throws IOException {
+    ComponentUpload component = createComponentUpload(WRONG_PACKAGE_PATH, PACKAGE_NAME);
+    checkValidationFailed(component, "Non-R archive extension or wrong upload path.");
```

Comment:
Minor: Checking on the exact failing message seems fairly brittle, at the very least I'd pull the message into some sort of shared variable (so it can be imported and reused in the tests, etc....)